crates/sc-subspace-block-relay/src/utils.rs

```diff
@@ -44,36 +44,42 @@ impl NetworkWrapper {
 }
 
 /// Network handle that allows making requests to specific peer and protocol.
+/// `RequestMsg` is the format of the request message sent on the wire.
 #[derive(Clone)]
-pub(crate) struct NetworkPeerHandle {
+pub(crate) struct NetworkPeerHandle<RequestMsg: Encode> {
     protocol_name: ProtocolName,
     who: PeerId,
     network: NetworkRequestService,
+    _p: std::marker::PhantomData<RequestMsg>,
 }
 
-impl NetworkPeerHandle {
+impl<RequestMsg: Encode> NetworkPeerHandle<RequestMsg> {
     fn new(protocol_name: ProtocolName, who: PeerId, network: NetworkRequestService) -> Self {
         Self {
             protocol_name,
             who,
             network,
+            _p: Default::default(),
         }
     }
 
-    /// Performs the request
+    /// Performs the request, where the request can be transformed into
+    /// the network format `RequestMsg`.
     pub(crate) async fn request<Request, Response>(
         &self,
         request: Request,
     ) -> Result<Response, RequestResponseErr>
     where
         Request: Encode,
         Response: Decode,
+        Request: Into<RequestMsg>,
     {
         let (tx, rx) = oneshot::channel();
+        let msg: RequestMsg = request.into();
```

Comment:
This is always more readable:
```suggestion
        let msg = RequestMsg::from(request);
```

Reply:
Changed this part, also removed Encode constraint as we discussed